back/app/controllers/web_api/v1/permissions_fields_controller.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class WebApi::V1::PermissionsFieldsController < ApplicationController
+  skip_after_action :verify_policy_scoped
+  before_action :set_permissions_field, only: %i[show update destroy reorder]
+
+  def index
+    authorize PermissionsField.new(permission: permission)
+
+    permissions_fields_service = Permissions::PermissionsFieldsService.new
+    if permissions_fields_service.custom_permitted_by_enabled?
+      # NEW version for verified actions
+      permissions_fields = permissions_fields_service.fields_for_permission(permission)
+      render json: WebApi::V1::PermissionsFieldSerializer.new(permissions_fields, params: jsonapi_serializer_params).serializable_hash
+    else
+      # Legacy version
+      permissions_fields = permission.permissions_fields.where(field_type: 'custom_field').order('custom_fields.ordering')
+      permissions_fields = paginate permissions_fields
+      permissions_fields = permissions_fields.includes(:custom_field)
+
+      render json: linked_json(
+        permissions_fields,
+        WebApi::V1::PermissionsFieldSerializer,
+        params: jsonapi_serializer_params,
+        include: %i[custom_field]
+      )
+    end
```

Comment:
Note: this file was renamed - this and the addition of a `reorder` endpoint is the only logic that has changed